File: stripe/src/main/java/com/stripe/android/AnalyticsRequest.java
package com.stripe.android;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.Map;

final class AnalyticsRequest {
    static final String HOST = "https://q.stripe.com";

    @NonNull
    static ApiRequest create(@NonNull Map<String, ?> params,
                             @NonNull ApiRequest.Options requestOptions) {
        return create(params, requestOptions, null);
    }

    @NonNull
    static ApiRequest create(@NonNull Map<String, ?> params,
                             @NonNull ApiRequest.Options requestOptions,
                             @Nullable AppInfo appInfo) {
        return new ApiRequest(StripeRequest.Method.GET, HOST, params, requestOptions,
                appInfo);
    }
}

File: stripe/src/main/java/com/stripe/android/AnalyticsRequest.kt
package com.stripe.android

internal data class AnalyticsRequest(
    override val params: Map<String, *>,
    internal val options: ApiRequest.Options,
    internal val appInfo: AppInfo? = null
) : StripeRequest() {
    override val method: Method = Method.GET
    override val baseUrl: String = HOST
    override val mimeType: MimeType = MimeType.Form
    override val headersFactory: RequestHeadersFactory = RequestHeadersFactory.Api(
        options = options,
        appInfo = appInfo
    )

    class Factory(
        private val logger: Logger = Logger.noop()
    ) {
        @JvmSynthetic
        internal fun create(
            params: Map<String, *>,
            options: ApiRequest.Options,
            appInfo: AppInfo? = null
        ): AnalyticsRequest {
            logger.info("Event: ${params[AnalyticsDataFactory.FIELD_EVENT]}")

            return AnalyticsRequest(
                params,
                options,
                appInfo
            )
        }
    }

    internal companion object {
        internal const val HOST = "https://q.stripe.com"
    }
}
